movieApp.js

```diff
@@ -1,170 +1,264 @@
-
-  const movieContainer = document.createElement("div");
-  movieContainer.className = "movie-container";
-
-  //
-  //  Create an img element and set its src attribute
-  const img = document.createElement("img");
-  img.width = 200;
-  img.src = movie.poster_url;
-
-  // Create a button element
-  const readMoreBtn = document.createElement("button");
-  readMoreBtn.textContent = "Read More";
-  readMoreBtn.className = "accordion";
-
-  // starRating
-  const starRating = document.createElement("div");
-  starRating.className = "star-rating";
-  const numberOfStars = 5;
-  let stars = [];
-
-  for (let x = 0; x < numberOfStars; x++) {
-    const star = document.createElement("i");
-    star.className = "rating__star far fa-star";
-    starRating.appendChild(star);
-    stars.push(star);
-  }
-  const starClassActive = "rating__star fas fa-star";
-  const starClassInactive = "rating__star far fa-star";
-  stars.forEach((star, clickedStarIndex) => {
-    star.onclick = () => {
-      stars.forEach((star, starIndex) => {
-        if (starIndex <= clickedStarIndex) {
-          star.className = starClassActive;
-        } else {
-          star.className = starClassInactive;
-        }
+document.addEventListener("DOMContentLoaded", getMoviesData);
+
+function getMoviesData() {
+  fetch(
+    "https://raw.githubusercontent.com/myselfankitaa/myselfankitaa.github.io/main/movies.json"
+  )
+    .then((moviesData) => {
+      return moviesData.json();
+    })
+    .then(processMoviesData)
+    .catch((error) => `ERROR !!!`)
+    .finally(() => console.log(`Done`));
+
+  function processMoviesData(movies) {
+    const nowStreaming = document.querySelector(".now-streaming");
+
+    function createMovieCard(movie) {
+      //Creat a cotainer to store the Image and readMore Button
+
+      const movieContainer = document.createElement("div");
+      movieContainer.className = "movie-container";
+
+      //
+      //  Create an img element and set its src attribute
+      const img = document.createElement("img");
+      img.width = 200;
+      img.src = movie.poster_url;
+
+      // Create a button element
+      const readMoreBtn = document.createElement("button");
+      readMoreBtn.textContent = "Read More";
+      readMoreBtn.className = "accordion";
+
+      // starRating
+      const starRating = document.createElement("div");
+      starRating.className = "star-rating";
+      const numberOfStars = 5;
+      let stars = [];
+
+      for (let x = 0; x < numberOfStars; x++) {
+        const star = document.createElement("i");
+        star.className = "rating__star far fa-star";
+        starRating.appendChild(star);
+        stars.push(star);
+      }
+      const starClassActive = "rating__star fas fa-star";
+      const starClassInactive = "rating__star far fa-star";
+      stars.forEach((star, clickedStarIndex) => {
+        star.onclick = () => {
+          stars.forEach((star, starIndex) => {
+            if (starIndex <= clickedStarIndex) {
```

Comment:
You can simplify this into one line :)

Reply:
I need a help to do so. I am trying but is not working out.

Reply:
still tried some changes